zk/realtime/relayer/streamer/server.go

```diff
@@ -158,13 +161,18 @@ func (server *StreamServer) relayRealtime() {
 	for {
 		select {
 		case <-server.ctx.Done():
+			log.Info("Server shutdown requested, notifying clients")
+			flag := StopFlag
+			server.BroadcastMessage(&flag, nil)
 			return
+
 		case rawMsg := <-rawMsgsChan:
-			// Broadcast ConsumerMessage directly
-			server.BroadcastMessage(rawMsg)
+			server.BroadcastMessage(nil, rawMsg)
```

Comment:
Nitpicking but i think it makes more sense to define serialization utility functions for specific typed messages. After which we encapsulate our broadcast of the serialized message to the ws clients
```suggestion
			bytes := fromKafkaMessage()
			server.BroadcastMessage(bytes)
```